### ISSUE TITLE
Laravel 5.4.19 - SQLSTATE[42S02]: Base table or view not found

### ISSUE THREAD

**Issue by patkruk** [yajra/laravel-oci8#279]
### Summary of problem or feature request

After upgrading Laravel to version 5.4.19, I started to receive "SQLSTATE[42S02]: Base table or view not found" errors. If I downgrade Laravel to 5.4.18, the issue goes away.

### System details

- Operating System: Ubuntu
- PHP Version: 7
- Laravel Version: 5.4.19
- Laravel-OCI8 Version: 5.4.7

**Comment by yajra** [comment]
It seems like there were changes on [Eloquent Builder](https://github.com/laravel/framework/commit/7b3bf0ba3a8b7c501a2d357afb00b1492b055c3c) that might cause your issue?

Can you please paste some snippets to reproduce the issue? Thanks!

**Comment by patkruk** [comment]
The issue occurs when I try to save a model, so in order to reproduce it, call the save() method on an Eloquent model. Reading data from the database is working fine.

**Comment by devbian** [comment]
@patkruk i got the same error. do you solved it?

**Comment by devbian** [comment]
i solve this error by rollback laravel to 5.4.10. 😁

**Comment by patkruk** [comment]
@devbian Same here. I just downgraded Laravel for now.

**Comment by patkruk** [comment]
@yajra Are you guys planning to investigate this issue?

**Steps To Reproduce:**

```
class Test extends \Eloquent {

protected $connection = 'non-default';
}

$test = Test::find($testID);
$test->prop = '123';

$test->save();
```

See this Laravel GitHub issue: https://github.com/laravel/framework/issues/19030

**Comment by ChaosPower** [comment]
Seems like the model instance got lost. Will check this out too.

**Comment by yajra** [comment]
@patkruk thanks for the info. Tried to replicate the issue but all is working fine on my side using `Laravel Framework 5.4.19`.

This is my setup, please confirm if correct?
1. Default DB connection is `mysql`
2. I created a model `Category` and set the connection to `oracle`.

```php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Category extends Model
{
    protected $connection = 'oracle';
}
```

Added route to test saving:
```php
use App\Category;
$router->get('test', function () {
    $category = Category::first();
    $category->title = "edited";
    $category->save();

    return $category;
});
```

**Comment by yajra** [comment]
@patkruk can you please confirm PR #285 if it fixes this issue? Got the issue hint from https://github.com/laravel/framework/pull/19049. Thanks for providing the details.

**Comment by wa7eedem** [comment]
I test the PR #285 and its fix the issue for me , thank you @yajra .

**Comment by ssbahamdain** [comment]
Thanks @yajra, we test #285 it's working perfectly 👍

**Comment by yajra** [comment]
PR released on v5.4.9. Thanks for testing 👍

**Comment by patkruk** [comment]
Yup, now it's working as expected. Thanks!
